File: Angle.java
public class Angle
{
    final public double a;

    public Angle(double a)
    {
        this.a = a;
    }

    public static Angle diff(Angle a1, Angle a2)
    {
        return new Angle(a1.a - a2.a);
    }

    public static Angle sum(Angle a1, Angle a2)
    {
        return new Angle(a1.a + a2.a);
    }

    public static Angle neg(Angle a)
    {
        return new Angle(-a.a);
    }

    public static Angle abs(Angle a)
    {
        if (a.a < 0)
            return new Angle(-a.a);
        else
            return new Angle(a.a);
    }

}

File: hyper-web/src/Angle.ts
export class Angle {
    readonly a: number;

    constructor(a: number) {
        this.a = a;
    }

    static diff(a1: Angle, a2: Angle): Angle {
        return new Angle(a1.a - a2.a);
    }

    static sum(a1: Angle, a2: Angle): Angle {
        return new Angle(a1.a + a2.a);
    }

    static neg(a: Angle): Angle {
        return new Angle(-a.a);
    }

    static abs(a: Angle): Angle {
        return new Angle(Math.abs(a.a));
    }
}